src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -1100,37 +1150,198 @@ private void PopulateParamsFromRestRequest(Dictionary<string, object?> parameter
+                    parameters: objectValueNodeInListValueNode));
             }
+        }
 
-            return role;
+        /// <summary>
+        /// Helper method to iterate over all the fields present in the input for the current field and add it to the dictionary
+        /// containing all entities and their corresponding fields.
+        /// </summary>
+        /// <param name="context">Middleware context.</param>
+        /// <param name="entityName">Name of the entity.</param>
+        /// <param name="schemaObject">Input object type for the field.</param>
+        /// <param name="fieldNodes">List of ObjectFieldNodes for the the input field.</param>
+        /// <param name="entityTofields">Dictionary to store all the entities and their corresponding fields referenced in the mutation.</param>
+        private void ProcessObjectFieldNodesForAuthZ(
+            IMiddlewareContext context,
+            string entityName,
+            InputObjectType schemaObject,
+            IReadOnlyList<ObjectFieldNode> fieldNodes,
+            Dictionary<string, HashSet<string>> entityTofields)
+        {
+            RuntimeConfig runtimeConfig = _runtimeConfigProvider.GetConfig();
+            entityTofields.TryAdd(entityName, new HashSet<string>());
+            string dataSourceName = GraphQLUtils.GetDataSourceNameFromGraphQLContext(context, runtimeConfig);
+            ISqlMetadataProvider metadataProvider = _sqlMetadataProviderFactory.GetMetadataProvider(dataSourceName);
+            foreach (ObjectFieldNode field in fieldNodes)
+            {
+                Tuple<IValueNode?, SyntaxKind> fieldDetails = GraphQLUtils.GetFieldDetails(field.Value, context.Variables);
+                SyntaxKind underlyingFieldKind = fieldDetails.Item2;
+
+                // For a column field, we do not have to recurse to process fields in the value - which is required for relationship fields.
+                if (GraphQLUtils.IsScalarField(underlyingFieldKind))
+                {
+                    // It might be the case that we are currently processing the fields for a linking input object.
+                    // Linking input objects enable users to provide input for fields belonging to the target entity and the linking entity.
+                    // Hence the backing column for fields belonging to the linking entity will not be present in the source definition of this target entity.
+                    // We need to skip such fields belonging to linking table as we do not perform authorization checks on them.
+                    if (metadataProvider.TryGetBackingColumn(entityName, field.Name.Value, out string? _))
+                    {
+                        // Only add those fields to this entity's set of fields which belong to this entity and not the linking entity.
+                        entityTofields[entityName].Add(field.Name.Value);
+                    }
+                }
+                else
+                {
```

Comment:
this else condition indicates field is a relationship field?

Reply:
yes